tests/test_cli_compile.py

```diff
@@ -939,6 +939,29 @@ def test_upgrade_packages_version_option_and_upgrade_no_existing_file(pip_conf,
     assert "small-fake-b==0.1" in out.stderr
 
 
+def test_upgrade_package_with_extra(pip_conf, make_package, runner):
+    """
+    piptools ignores extras on --upgrade-package/-P items if already constrained.
+    """
+    package_with_extra = make_package(
+        "package_with_extra",
+        extras_require={
+            "extra": ["small-fake-a==0.1"],
+        },
+    )
+
+    with open("requirements.in", "w") as req_in:
+        req_in.write(f"{package_with_extra}[extra]")
+
+    out = runner.invoke(
+        cli, ["--no-annotate", "--upgrade-package", "package_with_extra[extra]"]
+    )
+
+    assert out.exit_code == 0
+    assert "package_with_extra" in out.stderr
+    assert "small-fake-a==" in out.stderr
```

Comment:
Could you please update this to check the whole output from pip-compile ([example](https://github.com/jazzband/pip-tools/blob/6735e6271f489a73f490a1b28dd47f2616417e73/tests/test_cli_compile.py#L2567-L2601))?

Reply:
Updated as requested 😄